docs/usage.rst

```diff
@@ -0,0 +1,207 @@
+************
+Installation
+************
+
+.. code-block:: console
+
+   pip install flake8-async
+
+*****
+Usage
+*****
+
+install and run through flake8
+==============================
+
+.. code-block:: sh
+
+   pip install flake8 flake8-async
+   flake8 .
+
+install and run with pre-commit
+===============================
+
+If you use `pre-commit <https://pre-commit.com/>`_, you can use it with flake8-async by
+adding the following to your ``.pre-commit-config.yaml``:
+
+.. code-block:: yaml
+
+   minimum_pre_commit_version: '2.9.0'
+   repos:
+   - repo: https://github.com/python-trio/flake8-async
+     rev: 23.2.5
+     hooks:
+       - id: flake8-async
+         # args: [--enable=ASYNC, --disable=ASYNC9, --autofix=ASYNC]
+
+This is often considerably faster for large projects, because ``pre-commit``
+can avoid running ``flake8-async`` on unchanged files.
+Afterwards, run
+
+.. code-block:: sh
+
+   pip install pre-commit flake8-async
+   pre-commit run .
+
+install and run as standalone
+=============================
+
+If inside a git repository, running without arguments will run it against all ``*.py`` files in the repository.
+
+.. code-block:: sh
+
+   pip install flake8-async
+   flake8-async
+
+with autofixes
+--------------
+
+.. code-block:: sh
+
+   flake8-async --autofix=ASYNC
+
+specifying source files
+-----------------------
+
+.. code-block:: sh
+
+   flake8-async my_python_file.py
+
+zsh-only
+^^^^^^^^
+
+.. code-block:: zsh
+
+   flake8-async **/*.py
+
+
+Run through ruff
+================
+`Ruff <https://github.com/astral-sh/ruff>` is a linter and formatter that reimplements a lot of rules from various flake8 plugins. They currently only support a small subset of the rules though, see https://github.com/astral-sh/ruff/issues/8451 for current status and https://docs.astral.sh/ruff/rules/#flake8-async-async for documentation.
+
+*************
+Configuration
+*************
+
+`You can configure flake8 with command-line options <https://flake8.pycqa.org/en/latest/user/configuration.html>`_,
+but we prefer using a config file. The file needs to start with a section marker ``[flake8]`` and the following options are then parsed using flake8's config parser, and can be used just like any other flake8 options.
```

Comment:
Specify allowed filenames / types?

Reply:
rewrote/expanded it, referring to flake8 documentation for filenames/types/section marker. Also linking pyproject-flake8, and suggesting various ways of getting around us not having config support for running as a standalone.